system/lib/libc/musl/src/misc/emscripten_pthread.c

```diff
@@ -24,8 +24,10 @@ EM_JS(void, initPthreadsJS, (void), {
   PThread.initRuntime();
 })
 
+// This must run before any userland ctors
+// Note that ASan constructor priority is 1, and we must be higher.
 EMSCRIPTEN_KEEPALIVE
-__attribute__((constructor(99))) // This must run before any userland ctors
+__attribute__((constructor(0)))
```

Comment:
Before this we had 99 and 100, so there was an order, but this changes them both to 0 - isn't that wrong? I think we need to change asan's priority.

Reply:
The two constructors don't actually depend on each other. We only thought they did due to `pthread_cleanup`, but as it turns out, `pthread_cleanup` is purely JavsScript and does not care if the C side has been initialized.

Reply:
Fair point. However, I think it would be safer going forward to have a definite order there - I can see us adding more stuff to one of the constructors some day, and having flaky failures because the order is nondeterministic. So I think it's worth changing asan's priority, keeping all three in a clear order.

Reply:
Diff to change ASan's priority: https://reviews.llvm.org/D65684